pkg/tide/status.go

```diff
@@ -145,18 +146,35 @@ func requirementDiff(pr *PullRequest, q *config.TideQuery, cc contextChecker) (s
 	// Weight incorrect branches with very high diff so that we select the query
 	// for the correct branch.
 	targetBranchDenied := false
-	for _, excludedBranch := range q.ExcludedBranches {
-		if string(pr.BaseRef.Name) == excludedBranch {
+	if len(q.ExcludedBranches) > 0 {
+		branchExclusions, err := regexp.Compile(strings.Join(q.ExcludedBranches, `|`))
+		if err == nil && branchExclusions.MatchString(string(pr.BaseRef.Name)) {
 			targetBranchDenied = true
-			break
+		} else if err != nil {
+			// Fall back to exact matching if regex compilation fails
+			for _, excludedBranch := range q.ExcludedBranches {
+				if string(pr.BaseRef.Name) == excludedBranch {
+					targetBranchDenied = true
+					break
+				}
+			}
```

Comment:
Is it possible to extract a method to avoid the (mostly complete) duplication between this and the includedBranch logic?